### PR TITLE
Feature 5 vehicle list screen

### LINKED ISSUE #5: Vehicles screen
As a user I'd like to be able to list vehicles and see information about them

### PR DESCRIPTION
**Pull request by EVP1989**
Resolves #5

### CODE REVIEW

File: src/components/vehicle-list-screen.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        allowProps: true,
+        flex: 1,
+        type: "string",
+        renderHeader: params => {
+          return (
+            <Stack direction="row">
+              <Typography fontWeight={"bold"} fontSize={14}>{params.colDef.headerName}</Typography>
+            </Stack>
+          );
+        },
```
Comment (lauriLukkarinen) [inline src/components/vehicle-list-screen.tsx:50]:
```suggestion
        renderHeader: params => (
          <Stack direction="row">
            <Typography fontWeight={"bold"} fontSize={14}>{params.colDef.headerName}</Typography>
          </Stack>
        ),
```

File: src/components/vehicle-list-screen.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    row.id = truck.id!;
+    row.name = truck.plateNumber
+    row.number = truck.vin;
+    row.address = "-";
+    row.location = "-";
+    row.status = "-";
+    row.trailer = "-";
+    row.driver = "-";
+    return row;
+  };
```
Comment (lauriLukkarinen) [inline src/components/vehicle-list-screen.tsx:90]:
```suggestion
  const buildRow = (truck: Truck) => {
    return {
      id: truck.id,
      name: truck.plateNumber,
      number: truck.vin,
      address: "-",
      location: "-",
      status: "-",
      trailer: "-",
      driver: "-"
    } as const;
  };
```

File: src/components/vehicle-list-screen.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  /**
+   * Load data for the table
+   */
+  const loadData = async () => {
+    setLoading(true);
+
+    try {
+      const trucks: Truck[] = await trucksApi.listTrucks({});
```
Comment (lauriLukkarinen) [inline src/components/vehicle-list-screen.tsx:99]:
```suggestion
      const trucks = await trucksApi.listTrucks({});
```

File: src/components/vehicle-list-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+      const vehicleRows = trucks.map((truck) => buildRow(truck));
```
Comment (lauriLukkarinen) [inline src/components/vehicle-list-screen.tsx:100]:
```suggestion
      const vehicleRows = trucks.map(buildRow);
```

File: src/localization/locales/fi.json [hunk 5]
```diff
@@ -57,5 +57,24 @@
     "freights": {
       "title": "Kuormat"
     }
+  },
+  "errorHandling": {
+    "vehicleListing": "Ajoneuvolistaukssessa tapahtui virhe"
```
Comment (lauriLukkarinen) [inline src/localization/locales/fi.json:62]:
```suggestion
    "vehicleListing": "Ajoneuvolistauksessa tapahtui virhe"
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: open-api/typescript-client/apis.mustache
[the remark this answers is not in the document]
Comment (EVP1989) [reply open-api/typescript-client/apis.mustache:1]: This was some copypaste bloating, removed it.

**Review by lauriLukkarinen — APPROVED** [review]

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**